src/rdkafka.c

```diff
@@ -2609,13 +2609,18 @@ rd_kafka_consume_callback0 (rd_kafka_q_t *rkq, int timeout_ms, int max_cnt,
                                                 void *opaque),
                             void *opaque) {
         struct consume_ctx ctx = { .consume_cb = consume_cb, .opaque = opaque };
+        rd_kafka_op_res_t res;
 
-        rd_kafka_app_polled(rkq->rkq_rk);
+        if (timeout_ms)
+                rd_kafka_app_poll_blocking(rkq->rkq_rk);
+
+        res = rd_kafka_q_serve(rkq, timeout_ms, max_cnt,
+                               RD_KAFKA_Q_CB_RETURN,
+                               rd_kafka_consume_cb, &ctx);
 
-        return rd_kafka_q_serve(rkq, timeout_ms, max_cnt,
-                                RD_KAFKA_Q_CB_RETURN,
-                                rd_kafka_consume_cb, &ctx);
+        rd_kafka_app_polled(rkq->rkq_rk);
```

Comment:
Wouldn't this call to rd_kafka_app_polled be redundant now that `rd_kafka_q_serve` calls `rd_kafka_app_polled`()?

Reply:
s/q_serve/q_serve_rkmessages/

and yep, you are correct, fixed it :+1: